llvm/test/Transforms/LICM/iterate-over-dom-nodes.ll

```diff
@@ -0,0 +1,889 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -S -passes='cgscc(devirt<4>(inline,function<eager-inv;no-rerun>(early-cse<memssa>,simplifycfg<bonus-inst-threshold=1;no-forward-switch-cond;switch-range-to-icmp;no-switch-to-lookup;keep-loops;no-hoist-common-insts;no-hoist-loads-stores-with-cond-faulting;no-sink-common-insts;speculate-blocks;simplify-cond-branch;no-speculate-unpredictables>,loop-mssa(licm<allowspeculation>))))' < %s | FileCheck %s
```

Comment:
Should be able to get this down to one pass?

Reply:
I don't think so, not for this example. This is the best `reduce_pipeline.py` was able to do. In general this issue has been very hard to reproduce.

Reply:
You need to do some manual pipeline cleanup after reduce_pipeline. Things like the cgscc and devirt wrapper are usually not needed, as are most pass options.

Reply:
It's usually also possible to use some at least partially inlined IR.

Reply:
I shortened this line quite a bit. There are still four passes and the devirt. Anything less does not reproduce the error.